### DISCUSSION TITLE
Converting MDX to CJS for React Native can be challenging due to compatibility issues.

### DISCUSSION THREAD

**Discussion by zfben** [mdx-js/mdx#2450] [Q&A] [answered]
When I try to build cjs for node platform, `vfile` use `node:path`.
For browser platform, `decode-named-character-reference` use `document.createElement`.

Is there any way to solve it?

**Comment by ChristianMurphy** [comment, marked as answer]
Hey @zfben! 👋 
Sorry you ran into some confusion.

Taking a few steps back, setting aside the build questions.
MDX compiles to HTML tags by default, React Native does not support HTML tags.
You need something like https://github.com/danieldunderfelt/rn-mdx which translates the web elements into their React native equivalent.

Back to your build tool question, ESM and node built in are standard, they're supported by build tools.
Expo and react-native/babel-preset automatically take care of this for you, make sure you have one or both enabled https://reactnative.dev/docs/next/javascript-environment#javascript-syntax-transformers
If you still see an issue consider raising with the issue with your build tool provider.

`decode-named-character-reference` provides a react native implementation https://github.com/wooorm/decode-named-character-reference/blob/4347630741f82d7d4c7fb38763ba158951b2a1be/package.json#L35
`node:path` has several well tested and maintained polyfills for running on non-node environments like hermes.
Something in your build tool is not configured correctly or is broken.

**Reply by zfben** [reply]
Thank you for your answer, I built a custom mappings plugin for esbuild finally.
